File: Buy Sell Stock/solution_bruteforce.js
/**
 * Brute-force approach checks every possible buy/sell pair.
 * @param {number[]} prices
 * @returns {number}
 */
function maxProfitBruteforce(prices) {
  let best = 0;

  for (let buy = 0; buy < prices.length; buy += 1) {
    for (let sell = buy + 1; sell < prices.length; sell += 1) {
      const profit = prices[sell] - prices[buy];
      if (profit > best) {
        best = profit;
      }
    }
  }

  return best;
}

module.exports = { maxProfitBruteforce };

File: Buy Sell Stock/solution_bruteforce.py
from typing import List


def max_profit_bruteforce(prices: List[int]) -> int:
    """Return the max profit by checking every buy/sell pair."""

    best = 0

    for buy in range(len(prices)):
        for sell in range(buy + 1, len(prices)):
            profit = prices[sell] - prices[buy]
            if profit > best:
                best = profit

    return best
